src/core/os_windows.c

```diff
@@ -324,6 +324,39 @@ os_posix_fallocate(int fd, os_off_t offset, os_off_t len)
 	if (offset + len < offset)
 		return EFBIG;
 
+	HANDLE handle = (HANDLE)_get_osfhandle(fd);
+	if (handle == INVALID_HANDLE_VALUE) {
+		return -1;
```

Comment:
you need to map errors to those returned by the real posix_fallocate.

Reply:
done?